File: frontend/src/lib/types/event.ts
export interface Event {
    id: string;
    title: string;
    description: string;
    category_id: string;
    geometry: string; // GeoJSON string
    acquired: string;
    updated: string;
    source_url: string;
    severity: string;
    confidence: number;
    category_name?: string;
    category_color?: string;
    category_icon?: string;
}

export interface Category {
    id: string;
    name: string;
    description: string;
    color: string;
    icon: string;
}

export interface EventFilters {
    category?: string;
    severity?: string;
    limit?: number;
}

File: backend/ingestion-service/internal/models/event.go
package models

import (
	"time"
)

type Event struct {
	ID          string    `json:"id"`
	Title       string    `json:"title"`
	Description string    `json:"description"`
	CategoryID  string    `json:"category_id"`
	Latitude    float64   `json:"latitude"`
	Longitude   float64   `json:"longitude"`
	Geometry    string    `json:"geometry"` // GeoJSON string
	Acquired    time.Time `json:"acquired"`
	Updated     time.Time `json:"updated"`
	SourceURL   string    `json:"source_url"`
	Severity    string    `json:"severity"`
	Confidence  float64   `json:"confidence"`
	InsertedAt  time.Time `json:"inserted_at"`
}